base_classes/NXdetector.nxdl.xml

```diff
@@ -719,4 +719,24 @@
 		subdivision of a detector.
     </doc>
   </group>
+  <group name="pixel_shape" type="NXoff_geometry">
+    <doc>
+		  Shape description of each pixel. Use only if all pixels in the detector are of uniform shape.
+    </doc>
+  </group>
+  <group name="detector_shape" type="NXoff_geometry">
+    <doc>
+		  Shape description of the whole detector. Use only if pixels in the detector are not of uniform shape.
+    </doc>
+  </group>
+  <group name="pixel_shape" type="NXcylindrical_geometry">
```

Comment:
You cannot have two "pixel_shape" groups with the same name. For now this could be an either/or choice, but in the long run we did discuss some scheme of providing alternatives (OFF for the dumb programs, cylinders for the slightly smarter ones, and the CAD thing for whoever wants to implement it - or so). 
Probably worth thinking about that some more.